### PR TITLE
Cox Proportional Hazards: Enhancements

### PR DESCRIPTION
**Pull request by srikris**
Adde the following modules:

(1) Huber White estimator for the Cox Proprtional hazards

(2) Supporting stats (p-values & zStats) for the above

(3) Detailed design document for the Cox Proportional Hazards Module

(4) Design document for Huber-White Sandwich Estimator

### CODE REVIEW

**Comment by iyerr3** [comment]
Krishna (@srikris), 
the design documents look good; please consider adding references and pseudo-code. Pseudo-code immensely helps in understanding the code.

File: doc/literature.bib [hunk 1]
```diff
@@ -432,10 +472,9 @@ @article{hager2006survey
...
-	Title= {Regression models and life-tables},
-	Year = {1972},
 	Journal = {Journal of the Royal Statistical Society. Series B (Methodological)},
-	Volume = {34},
 	Number = {2},
 	Pages = {187-220},
-}
+	Title = {Regression models and life-tables},
+	Volume = {34},
+	Year = {1972}}
```
Comment (iyerr3) [inline doc/literature.bib:480]: (@ commiter) Please ignore bib file during merge since the changes are only present due to use of BibDesk.

**Comment by iyerr3** [comment]
@srikris There is quite a bit of overlap in the code for robust_cox  vs cox_prop (primarily because of Newton's method being repeated). Any plans of abstracting the Newton's method so robust cox only includes the HB estimation?

**Comment by iyerr3** [comment]
I am getting the following error on running your test: 

psql:/tmp/madlib.VG2nIv/stats/test/cox_proportional_hazards.sql_in.tmp:151: 
ERROR:  column "std_err" does not exist
LINE 6:     relative_error(std_err, ARRAY[0.5844,0.3207]) < 1e-2 AND

This is only for the robust_cox_prop_hazards version.

File: src/ports/postgres/modules/stats/cox_prop_hazards.py_in [hunk 2]
```diff
@@ -216,3 +216,30 @@ def compute_cox_prop_hazards(MADlibSchema, source, indepColumn,
...
+			(_cox_intermediate_state).x,
+			_cox_dep_column,
+			(_cox_intermediate_state).exp_coef_x,
+			(_cox_intermediate_state).x_exp_coef_x,
+			(_cox_intermediate_state).x_xTrans_exp_coef_x,
+			NULL ORDER by _cox_dep_column DESC
+		)).* FROM
+		(
+			SELECT {depColumn} as _cox_dep_column,
+					 intermediate_cox_prop_hazards({indepColumn}, ARRAY{coef})
```
Comment (iyerr3) [inline src/ports/postgres/modules/stats/cox_prop_hazards.py_in:235]: I get the following when I enter a subquery from your test: 

ERROR:  plpy.SPIError: function intermediate_cox_prop_hazards(double precision[], numeric[]) does not exist (plpython.c:4648)
LINE 13:         intermediate_cox_prop_hazards(ARRAY[grp, wbc], ARRAY[2...
HINT:  No function matches the given name and argument types. You may need to add explicit type casts.

**Comment by cwelton** [comment]
1 + 2 still have outstanding work that needs to be resolved.

Design doc work has largely already been pulled.

It is unlikely that this PR will be pulled as it currently stands, but rather work through the remaining issues.

**Comment by haying** [comment]
This pull request should be closed upon 1.4 release.

**Pull request closed without merging**